### PR TITLE
Docs: update Full Text Search doc

### PR DESCRIPTION
**Pull request by monicakh**
## What kind of change does this PR introduce?

doc update...

### CODE REVIEW

**Comment by monicakh** [comment]
@charislam Can you also double check the syntax for Kotlin library?

**Review by fsansalvadore — APPROVED** [review]

File: apps/docs/content/guides/database/full-text-search.mdx [hunk 1]
```diff
@@ -497,6 +497,77 @@ val data = supabase.from("books").select {
 
 Notice how searching for `big` includes results with the word `bigger` (or `biggest`, etc).
 
+## Partial search
+
+Partial search is particularly useful when you want to find matches on substrings within your data.
+
+### Implementing partial search
+
+You can use the `:*` syntax with to_tsquery(). Here's an example that searches for any book titles beginning with "Lit":
```
Comment (charislam) [inline apps/docs/content/guides/database/full-text-search.mdx:506]:
```suggestion
You can use the `:*` syntax with `to_tsquery()`. Here's an example that searches for any book titles beginning with "Lit":
```

File: apps/docs/content/guides/database/full-text-search.mdx [hunk 2]
```diff
@@ -497,6 +497,77 @@ val data = supabase.from("books").select {
...
+
+```sql
+SELECT title FROM books WHERE to_tsvector(title) @@ to_tsquery('Lit:*');
```
Comment (charislam) [inline apps/docs/content/guides/database/full-text-search.mdx:509]:
```suggestion
select title from books where to_tsvector(title) @@ to_tsquery('Lit:*');
```

just for consistency

File: apps/docs/content/guides/database/full-text-search.mdx [hunk 3]
```diff
@@ -497,6 +497,77 @@ val data = supabase.from("books").select {
...
+>
+<TabPanel id="sql" label="SQL">
+
+```sql
+CREATE OR REPLACE FUNCTION search_books_by_title_prefix(prefix text)
+RETURNS SETOF books AS $$
+BEGIN
+  RETURN QUERY SELECT * FROM books WHERE to_tsvector('english', title) @@ to_tsquery(prefix || ':*');
+END;
+$$ LANGUAGE plpgsql;
```
Comment (charislam) [inline apps/docs/content/guides/database/full-text-search.mdx:533]:
```suggestion
create or replace function search_books_by_title_prefix(prefix text)
return setof books AS $$
begin
  return query
  select * from books where to_tsvector('english', title) @@ to_tsquery(prefix || ':*');
end;
$$ language plpgsql;
```

File: apps/docs/content/guides/database/full-text-search.mdx [hunk 4]
```diff
@@ -497,6 +497,77 @@ val data = supabase.from("books").select {
...
+
+```kotlin
+val rpcParams = mapOf("prefix" to "Lit")
+val result = supabase.postgrest.rpc("search_books_by_title_prefix", rpcParams)
+```
+
+</TabPanel>
+</Tabs>
+
+This function takes a prefix parameter and returns all books where the title contains a word starting with that prefix. The `:*` operator is used to denote a prefix match in the to_tsquery() function.
```
Comment (charislam) [inline apps/docs/content/guides/database/full-text-search.mdx:561]:
```suggestion
This function takes a prefix parameter and returns all books where the title contains a word starting with that prefix. The `:*` operator is used to denote a prefix match in the `to_tsquery()` function.
```

**Review by charislam — APPROVED** [review]

**Pull request merged**